pennylane/beta/tapes/tape.py

```diff
@@ -936,3 +936,387 @@ def execute_device(self, params, device):
+                the current tape parameter values are used (via :meth:`~.get_parameters`).
+
+        Keyword Args:
+            method="best" (str): The differentiation method. Must be one of ``"numeric"``,
+                ``"analytic"``, ``"best"``, or ``"device"``.
+            h=1e-7 (float): finite difference method step size
+            order=1 (int): The order of the finite difference method to use. ``1`` corresponds
+                to forward finite differences, ``2`` to centered finite differences.
+
+        Returns:
+            array[float]: 2-dimensional array of shape ``(tape.num_params, tape.output_dim)``
+
+        **Example**
+
+        .. code-block:: python
+
+            from pennylane.beta.tapes import QuantumTape
+            from pennylane.beta.queuing import expval, var, sample, probs
+
+            with QuantumTape() as tape:
+                qml.RX(0.432, wires=0)
+                qml.RY(0.543, wires=0)
+                qml.CNOT(wires=[0, 'a'])
+                qml.RX(0.133, wires='a')
+                probs(wires=[0, 'a'])
+
+        If parameters are not provided, the existing tape parameters are used:
+
+        >>> dev = qml.device("default.qubit", wires=[0, 'a'])
+        >>> tape.jacobian(dev)
+        array([[-0.178441  , -0.23358253, -0.05892804],
+               [-0.00079144, -0.00103601,  0.05892804],
+               [ 0.00079144,  0.00103601,  0.00737611],
+               [ 0.178441  ,  0.23358253, -0.00737611]])
+
+        Parameters can be optionally passed during execution:
+
+        >>> tape.jacobian(dev, params=[1.0, 0.0, 1.0])
+        array([[-3.24029934e-01, -9.99200722e-09, -3.24029934e-01],
+               [-9.67055711e-02, -2.77555756e-09,  3.24029935e-01],
+               [ 9.67055709e-02,  3.05311332e-09,  9.67055709e-02],
+               [ 3.24029935e-01,  1.08246745e-08, -9.67055711e-02]])
+
+        Parameters provided for execution are temporary, and do not affect
+        the tapes' parameters in-place:
+
+        >>> tape.get_parameters()
+        [0.432, 0.543, 0.133]
+
+        Explicitly setting the trainable parameters can significantly reduce
+        computational resources, as non-trainable parameters are ignored
+        during the computation:
+
+        >>> tape.trainable_params = {0} # set only the first parameter as trainable
+        >>> tape.jacobian(dev)
+        array([[-0.178441  ],
+               [-0.00079144],
+               [ 0.00079144],
+               [ 0.178441  ]])
+
+        If a tape has no trainable parameters, the Jacobian will be empty:
+
+        >>> tape.trainable_params = {}
+        >>> tape.jacobian(dev)
+        array([], shape=(4, 0), dtype=float64)
+        """
+        method = options.get("method", "best")
+
+        if method not in ("best", "numeric", "analytic", "device"):
+            raise ValueError(f"Unknown gradient method '{method}'")
+
+        if params is None:
+            params = self.get_parameters()
```

Comment:
This is so nice, we can even make tapes one day that are never given parameters by the user, like in neural networks!